File: app/src/main/java/net/opendasharchive/openarchive/util/SelectiveViewPager.java
package net.opendasharchive.openarchive.util;

import android.content.Context;
import android.util.AttributeSet;
import android.view.MotionEvent;

import androidx.viewpager.widget.ViewPager;

public class SelectiveViewPager extends ViewPager {
    private boolean paging = true;


    public SelectiveViewPager(Context context) {
        super(context);
    }

    public SelectiveViewPager(Context context, AttributeSet attributeSet){
        super(context, attributeSet);
    }

    @Override
    public boolean onInterceptTouchEvent(MotionEvent e) {
        if (paging) {
            return super.onInterceptTouchEvent(e);
        }

        return false;
    }

    @Override
    public boolean onTouchEvent(MotionEvent ev) {
        return super.onTouchEvent(ev);
    }

    public void setPaging(boolean p){ paging = p; }

}

File: app/src/main/java/net/opendasharchive/openarchive/util/SelectiveViewPager.kt
package net.opendasharchive.openarchive.util

import android.content.Context
import android.util.AttributeSet
import android.view.MotionEvent
import androidx.viewpager.widget.ViewPager

class SelectiveViewPager : ViewPager {

    constructor(context: Context): super(context)
    constructor(context: Context, attributeSet: AttributeSet): super(context, attributeSet)

    private var paging = true

    override fun onInterceptTouchEvent(e: MotionEvent?): Boolean {
        return if (paging) {
            super.onInterceptTouchEvent(e)
        } else false
    }

    override fun onTouchEvent(ev: MotionEvent?): Boolean {
        return super.onTouchEvent(ev)
    }

}